src/Polysemy/Internal.hs

```diff
@@ -130,6 +130,43 @@ import Polysemy.Internal.Union
 -- Notice that we must use @-XTypeApplications@ to specify that we'd like to
 -- use the ('Polysemy.Output.Output' 'Int') effect.
 --
+-- Note:
+-- Don't use type aliases to distinguish between different types of
+-- effects, the differences are erased at compile-time. The following
+-- will not work as you might expect:
+--
+-- @
+-- type A = String
+-- type B = String
+--
+-- foo
+--   :: ( Member (Input A) r
```

Comment:
In all of your code samples, please change:

* `Member` => `'Member'`
* `Input` => `'Polysemy.Input.Input'`
* `input` => `'Polysemy.Input.input'`
* `Sem` => `'Sem'`

which will make the haddocks link properly.